### ISSUE TITLE
(C#) exceptions do not copy the response body into exceptions

### ISSUE THREAD

**Issue by koolraap** [Azure/autorest#693]
If I have a C# WebApi controller return a BadRequestErrorMessageResult (as an example, it's valid to return a body with other status codes)
```csharp
       [SwaggerResponse(HttpStatusCode.OK, "PutWidgetResponse", typeof(ApiResponse<WidgetModel>))]
        public async Task<IHttpActionResult> PutWidget(int id, [FromBody] WidgetModel widget)
        {
            // Validate
            if (id <= 0)
            {
                return BadRequest("id={0} is invalid", id);
            }
            // etc
```
The response body has the following JSON (from memory, it's approximately this):
```csharp
{ "message" : "id=0 is invalid"}
```

The generated C# proxy (I've removed a few irrelevant lines from this snippet) sets the response body to null:
```csharp
            _httpResponse = await this.Client.HttpClient.SendAsync(_httpRequest, cancellationToken).ConfigureAwait(false);
            HttpStatusCode _statusCode = _httpResponse.StatusCode;
            string _responseContent = null;
            if ((int)_statusCode != 200)
            {
                var ex = new HttpOperationException(string.Format("Operation returned an invalid status code '{0}'", _statusCode));
                ex.Request = new HttpRequestMessageWrapper(_httpRequest, _requestContent);
                ex.Response = new HttpResponseMessageWrapper(_httpResponse, _responseContent);
```

In my use case it'd be great to have the Response body get copied into the exception so we can mine it for more useful information.

Is it possible to get this changed so the response body is copied into the exception?

**Comment by stankovski** [comment]
If your swagger spec defines a model for the "default" status (see [error modeling](https://github.com/Azure/autorest/blob/master/Documentation/defining-clients-swagger.md#error-modeling)), you will get a strongly typed body for your exception.

**Comment by koolraap** [comment]
**TL;DR** garbage in==garbage out.

Thanks, that clears it up. It was user error on my part and non-obvious Swashbuckle behaviour.

Initially I'd avoided decorating my controller methods with this...
    `[SwaggerResponse(HttpStatusCode.Unauthorized, "", typeof(Exception))]` 
...as it was generating `object` in Autorest. Turns out Swashbuckle doesn't have XML documentation available for System.Exception so it doesn't emit anything useful.

Changing to this...
    `[SwaggerResponse(HttpStatusCode.Unauthorized, "", typeof(MyExceptionModel))]` 
Autorest works as expected.

**Comment by koolraap** [comment]
Actually the more I think about it, I think not copying the response body into the exception is a bug. It throws away information that may be useful.

**Comment by stankovski** [comment]
The response body is not getting copied into the exception probably because your swagger doesn't have any response payloads modelled for the operation. In that case we assume there is no body.

**Comment by koolraap** [comment]
@stankovski: Correct vs useful :-) I certainly understand the decision, but personally I'd err in favour of helping consumers than not. Setting the body to null punishes the consumer for a swagger mistake (omitting all possible HTTP status codes) which is beyond their control.

**Comment by brentonw** [comment]
@stankovski - so what happens when you can't control the Swagger?

Why not just set the body and allow the caller of the generated code to access anything they need?
